controllers/securitypolicy_controller.go

```diff
@@ -195,6 +195,15 @@ func (r *SecurityPolicyReconciler) delete(ctx context.Context, policy *tykv1.Sec
 			return nil
 		}
 
+		// At the moment, the Policy API returns 500 instead of 404 in cases of deleting non-existent Policies.
+		// So that reconciliation fails since non-existence couldn't be understood by Operator. Until this
+		// issue is fixed on Gateway level, add an ad-hoc API call to verify that policy's existence on Tyk level.
+		// If the Policy does not exist, no need to reconcile again. Otherwise, return err and reconcile.
+		_, errTyk := klient.Universal.Portal().Policy().Get(ctx, policy.Status.PolID)
+		if opclient.IsNotFound(errTyk) {
+			return nil
```

Comment:
Instead of returning here, we should continue to perform operations defined below if.
```
err := r.updateStatusOfLinkedAPIs(ctx, policy, true)
	if err != nil {
		return err
	}

	err = klient.Universal.HotReload(ctx)
	if err != nil {
		r.Log.Error(err, "Failed to hot-reload Tyk after deleting a Policy",
			"Policy", client.ObjectKeyFromObject(policy),
		)

		return err
	}
```

Reply:
I updated can you please re-review again? Now, the execution continues only if `errTyk` is `404`. The other response codes 200 or even 500 make operator continue working.